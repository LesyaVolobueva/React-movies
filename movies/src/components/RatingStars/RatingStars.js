import React from 'react';
import 'font-awesome/css/font-awesome.min.css';
import styles from './ratingStars.module.scss';
import PropTypes from 'prop-types';

class RatingStars extends React.Component {
  render() {
    const { setRating, stars, id } = this.props;
    const starItems = [];
    const starsCount = 5;

    for (let i = 1; i <= starsCount; i = i + 1) {
      let styleClass = styles.star;

      if (stars && stars >= i) {
        styleClass = styles.starActive;
      }

      starItems.push(
        <label
          className={styleClass}
          key={i}
          onClick={setRating.bind(null, id, i)}
        >
          â˜…
        </label>
      );
    }

    return(
      <div className={styles.rate}>
        {starItems}
      </div>
    );
  }
}

RatingStars.propTypes = {
  setRating: PropTypes.func,
  stars: PropTypes.number,
  id: PropTypes.number,
};

export default RatingStars;
