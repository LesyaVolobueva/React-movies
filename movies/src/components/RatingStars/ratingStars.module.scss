.rate {
  flex-basis: 100%;
  text-align: center;
  padding: 5px;

  &:hover label {
    color: #ffd700;
  }
}

.star,
.starActive {
  display: inline-block;
  padding: 3px;
  vertical-align: middle;
  line-height: 1;
  font-size: 20px;
  color: #ababab;
  transition: color 0.2s ease-out;
  cursor: pointer;
}

.rate label:hover ~ label {
  color: #ababab;
}

.starActive {
  color: #ffd700;
}
