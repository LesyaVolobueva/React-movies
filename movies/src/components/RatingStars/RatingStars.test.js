import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import RatingStars from './RatingStars';

Enzyme.configure({ adapter: new Adapter() });

describe('-- RatingStars --', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <RatingStars setRating={() => {}}/>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('setRating function is called on submit', () => {
    const mockedFunc = jest.fn();

    const wrapper = mount(
      <MemoryRouter>
        <RatingStars setRating={mockedFunc}/>
      </MemoryRouter>
    );

    expect(mockedFunc.mock.calls.length).toBe(0);

    wrapper.find('label').first().simulate('click');

    expect(mockedFunc.mock.calls.length).toBe(1);
  });
});
