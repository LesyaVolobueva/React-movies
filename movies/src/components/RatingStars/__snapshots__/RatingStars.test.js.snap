// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-- RatingStars -- renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    onClick={[Function]}
  >
    ★
  </label>
  <label
    className={undefined}
    onClick={[Function]}
  >
    ★
  </label>
  <label
    className={undefined}
    onClick={[Function]}
  >
    ★
  </label>
  <label
    className={undefined}
    onClick={[Function]}
  >
    ★
  </label>
  <label
    className={undefined}
    onClick={[Function]}
  >
    ★
  </label>
</div>
`;
